azurerm/data_source_kubernetes_cluster.go

```diff
@@ -0,0 +1,358 @@
+			return fmt.Errorf("Error setting `service_principal`: %+v", err)
+		}
+	}
+
+	kubeConfigRaw, kubeConfig := flattenKubernetesClusterDataSourceAccessProfile(&profile)
+	d.Set("kube_config_raw", kubeConfigRaw)
+
+	if err := d.Set("kube_config", kubeConfig); err != nil {
+		return fmt.Errorf("Error setting `kube_config`: %+v", err)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func flattenKubernetesClusterDataSourceLinuxProfile(input *containerservice.LinuxProfile) []interface{} {
+	values := make(map[string]interface{})
+	sshKeys := make([]interface{}, 0)
+
+	if profile := input; profile != nil {
+		if username := profile.AdminUsername; username != nil {
+			values["admin_username"] = *username
+		}
+
+		if ssh := profile.SSH; ssh != nil {
+			if keys := ssh.PublicKeys; keys != nil {
+				for _, sshKey := range *keys {
+					if keyData := sshKey.KeyData; keyData != nil {
+						outputs := make(map[string]interface{}, 0)
+						outputs["key_data"] = *keyData
+						sshKeys = append(sshKeys, outputs)
+					}
+				}
+			}
+		}
+	}
+
+	values["ssh_key"] = sshKeys
+
+	return []interface{}{values}
+}
+
+func flattenKubernetesClusterDataSourceAgentPoolProfiles(input *[]containerservice.AgentPoolProfile) []interface{} {
+	agentPoolProfiles := make([]interface{}, 0)
+
+	if profiles := input; profiles != nil {
+		for _, profile := range *profiles {
+			agentPoolProfile := make(map[string]interface{})
+
+			if profile.Count != nil {
+				agentPoolProfile["count"] = int(*profile.Count)
+			}
+
+			if profile.DNSPrefix != nil {
+				agentPoolProfile["dns_prefix"] = *profile.DNSPrefix
+			}
+
+			if profile.Name != nil {
+				agentPoolProfile["name"] = *profile.Name
+			}
+
+			if profile.VMSize != "" {
+				agentPoolProfile["vm_size"] = string(profile.VMSize)
+			}
+
+			if profile.OsDiskSizeGB != nil {
+				agentPoolProfile["os_disk_size_gb"] = int(*profile.OsDiskSizeGB)
+			}
+
+			if profile.VnetSubnetID != nil {
+				agentPoolProfile["vnet_subnet_id"] = *profile.VnetSubnetID
+			}
+
+			if profile.OsType != "" {
+				agentPoolProfile["os_type"] = string(profile.OsType)
+			}
+
+			agentPoolProfiles = append(agentPoolProfiles, agentPoolProfile)
+		}
+	}
+
+	return agentPoolProfiles
+}
+
+func flattenKubernetesClusterDataSourceServicePrincipalProfile(profile *containerservice.ServicePrincipalProfile) *schema.Set {
+	if profile == nil {
+		return nil
+	}
+
+	servicePrincipalProfiles := &schema.Set{
+		F: resourceAzureRMKubernetesClusterServicePrincipalProfileHash,
+	}
+
+	values := make(map[string]interface{})
+
+	if clientId := profile.ClientID; clientId != nil {
+		values["client_id"] = *clientId
+	}
+
+	if secret := profile.Secret; secret != nil {
+		values["client_secret"] = *secret
+	}
+
+	servicePrincipalProfiles.Add(values)
+
+	return servicePrincipalProfiles
+}
+
+func flattenKubernetesClusterDataSourceAccessProfile(profile *containerservice.ManagedClusterAccessProfile) (*string, []interface{}) {
+	if profile != nil {
```

Comment:
Could be inverted to return `return nil, []interface{}{}` if nil and to reduce nesting.

Reply:
👍 fixed